remote/remote_state.go

```diff
@@ -121,7 +123,16 @@ func (remoteState *RemoteState) differsFrom(existingBackend *TerraformBackend, t
 // Convert the RemoteState config into the format used by the terraform init command
 func (remoteState RemoteState) ToTerraformInitArgs() []string {
 	backendConfigArgs := []string{}
-	for key, value := range remoteState.Config {
+
+	initializer, hasInitializer := remoteStateInitializers[remoteState.Backend]
+	if !hasInitializer {
+		// return empty array
+		return backendConfigArgs
```

Comment:
Doesn't seem backwards compatible to return an empty array here. Instead, you should loop over `remoteState.Config` as before to fill `backendConfigArgs`. Or, more accurately, you should set `config = remoteState.Config` by default and then override it with `initializer.GetTerraformInitArgs` if `initializer` exists.

Reply:
True, fixed and also wrote a test for that.